azure-pipelines/flake8.yml

```diff
@@ -5,20 +5,26 @@
 # this software in either electronic or hard copy form.
 #
 
-pool:
-  vmImage: 'Ubuntu 16.04'
+jobs:
 
-steps:
-- task: UsePythonVersion@0
-  inputs:
-    versionSpec: 2.7
-    architecture: 'x64'
+- job: ${{ parameters.name }}
+  pool:
+    vmImage: 'ubuntu-20.04'
 
-- script: |
-    python -m pip install --upgrade pip setuptools wheel
-  displayName: 'Install prerequisites'
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: 3.9
+      addToPath: True
+      architecture: 'x64'
 
-- script: |
-    python -m pip install flake8
-    flake8 .
-  displayName: 'Run flake8'
+  - script: |
+      python -m pip install --upgrade pip setuptools wheel
+      pip install -r requirements.txt
+      pip install -r tests/requirements.txt
+    displayName: Install dependencies
+
+  - script: |
+      python -m pip install flake8
```

Comment:
Why do you sometimes use `pip` and sometimes use `python -m pip`?

Also, in my experience, it's better to install all the requirements in the same command line:
```yml
  - script: |
      pip install --upgrade pip setuptools wheel
      pip install --upgrade -r requirements.txt -r tests/requirements.txt flake8
    displayName: Install dependencies

  - script: flake8 .
    displayName: 'Run flake8'
```

This way, if there is any exclusion between all these requirements, the command will fail instead of uninstalling previously installed deps.

Reply:
That's legacy code from this repository. I also like to install everything in the same command. I can refactor this.